examples/sampling-slice-adaptive-hyperrectangles-mcmc.ipynb

```diff
@@ -0,0 +1,948 @@
6R0SyWUJHzcnMYiqRs7oXFGDUVOweJskIJEGAlixO5uLf56SGoyEVXZDQ9HsWj6kKgqdSpUjI/7SftrSiIqTMZCjtbr5yER+u2JhFgVCaBpfo5UFjcGaDUcHHLtyyZ8wY4IfyoQhvfPAWc9NgV3qWJDqxqrsewR5hdWe+hRMKJnRzrixD1qJT6PKhIhii0/HgtmRXNme/xdGefnAyaYxFcYRT1nYrTmEBk52u6tMWVkYJsfdbzsipgZqKUhal+EiWLCo7BOCIf8z4IYIQEoyRZIsvH1CAWpAfxaWkCSqHLqxXzei1NdCAtqdeHRYLFlDgflIfV3TaUyLDA+JyMVYTYiWi6UIlPKhqVDm0B5mf8BduKuRrOuSyCUspDQojLgEvRb+VJKR3yhU4dpzN2vCNSyi3AFoB169bNrkOli4tB0Yx/R15jxeJ5zvtcXFxONxkhhFm4BiFENc7ljU474aTC/JSzkKC7fdm3DYViLPLrFg5TuSo4TgkAOYEu/we00TOKKjVWBA/zlpKiVM2AZZU5IRQ82YPGFlSC8XS2aDDA/GhnoduZFDblTLG4KwkpOSt+YswxAHxaypbIoCeYsI1jjd0w3akE+gq3eQpObmBO5ITlQoRU8UjVFj/l1DYQz6B5oNxjd7ucnxlBUNw65FxSGlvqcpNknpXJLxIsD9cQL/GSGUdBSsrilo600PAW3Ts2FYFdtIlcnJJATyVu6qoqVrc6Xbk3lZLReAYEHBB9CLmk6BgZrVCJUPM+3h1Zt8tiNqdCRL7iJAoV0e7kfqpjTQWfClVIW7FrKyVanMr0kMMeva6cv2K1bVtXMA4eva93PL2OxzkxTwmM2yY5xrMH+j1xLLWgpnKWLdu1zimWfanYxAOSHDC/F3KWK/3V/HShYmUf3SR3V+aiiyDoLhSrjGOuFkIgpXxykuMNmq5/QohzAfPJ6gFb5s7lQF/B0S4uLi4u08aqVas4duzY+A0/OPwIeBE4SwjxIPBXwLdP75Sc0QVqmX1tpWs0RmmZVWKRBHubyWTGXqXWgGXxVuIlS83DdIHbQrGkD9Y+QI9LSqsapV4PJelQQbtEnuCfTKUpy2szIhLMUwz3wzyr0OJkNxPlsGcg+3qs2ccsK+P+WBoj4/SESVBc+Uh5F7AgNVCwPZywr0UrmkZSUWyKTFLRwDu19eL6kaAtPfpYxIus6usYAvE403BINDchDnjsIl2LCNhk7qDIPbsSSbsSolHYrWDWeeZT7vPYlHXzPJS89qaL42TE7PPy3FQ9FOoL89P+MZO9tAwXxjcBVKSdzhFK1Rhezf7sREXx+KRTJd/imM9QOFlg5a2ONcPCnO1ZQaNETTIvEyJaWg1AXCjsERP/LI+F+X1oPoOmu7Fj2zlgfpmQgiWEeApYDdRBVs2VwGQVrN8BXwT+X+PvS5bt9wshngPWAyE3/srlTOVMqmc11lynyxJXEDfQbuk3UCiYubjMFFLKp4UQNcAn0GWs/y6lHN9MMsfQpD3l9fz0CPMzw2McoSORzFOCOQXLQVRrFyEWkpqQEDcYSbFi8TxK06Ei0RQ5nKwzflH8+2cs4WmqmOfULPQV/QXpQoVoLEzXLqdro3jKHPeMF9dVMDkHlodrSKk4Omrt9uguiQtladFU82ca7yUHUTTJgCh0Z5ynFK+3Z1VIcjFP9oQMFLSYGp48FSs/QYYVVUr93kzSeHJ2zP7VlFv8mF3tIZVRHV1oy9QwaOU2ZXZJUn8eJ/L5nawtaeI2R3tbe8KLuWfBWgd8aDKFGYUQz6IntKgSQvQA30VXrH4jhPgHoAv4a6P5K+i1tlqAOPD3Ex3HxcXFxcVlIgghnpJSfgE46bDtjCH/h3hJsnNKxznVvQEIy8kqN+OLBqVqbMz9sykymhn0KjLFV+0nOx9rFjkrsbx4krGlqOLC38JE65iZxXx4GI46x1dNBNP6Mhv3YbwxpjIHKQtT1ENxi2yB298kEGqacum13S7PGP0pqjaGcjBxgd90wZvI9ZnO+5hvjS6G1UVwMklEJkpAjG2ht3GmWLCAY8A5OCSdKIaU8vNFdn3Coa0E/mmifbu4uMw8s2HdcnGZZa6wvjHisa49TXMZB0GMDL/ztBTsmcxKbl6XAJQYRWz9sZRtl8y+nnj/oUSGqOIrcP+bNHNAILLSM8b3n9Nas5AaFFGyrGQ0CcLpZMe+AKVqgsQYsviog7Vnasz8jRj3+ZVMycerXCn0iLBmQ5wuPELgLVCMiisho/EMxSLHSrWxFx6smMpiflyZE5NNLDFVIkkla02fyLyszOQcrTMx71QwnqacSShpp8hEFawqoEEIcRBys5NS3j4js3JxcZnTnElukC4uQohvAv8CzBNChMn95qYxkibNJUwBPulb4Lwq7yigj49Zu8cpO59VIPFpiaLZ7PIJpxTKcI4vmQxzTL8aE6e5OmUyK4aTO5lnnCQDUSZfA2symNa82YhdySUrcN4/FEmybP7kkntONFHJdJE/9fE+L1NeFLFgxtpNypIzJSY+V6ursjXerUwtLIae3/dEMneeKtbslP2hJLHYxL7XpoOJKlj/OpOTcHFxcTmT2Fy3eVr7u++q+8Zt8+CDD/LMM8/g9XrxeDw89thjrF+/flrnYWXjxo08/PDDrFu3bsp9vPTSS3z729/G4/Hg8/l49NFHCwoZzwZSyoeAh4QQD0kpJ533XwixAj3m+Bx0XWSLlPKH0zzNLJo0UjFXXFQQYH8qtGWLyubSPTvRk2ma9eIxYydgmGPMgFzoc8iAZ2WqSvVkyXdpnAnGUzYymmQgMnuWhslyMDXIgGdyFjJApNwAACAASURBVMPUOAr0RDBrPM1VmhyTkpxedolOFqEr65WhJjzLzpm1sSeapn2vEGIlcImUcpcQogKmnK3TxcXFBYDNwaPjthlf9fjgs3//fnbu3EltbS1lZWX4/X7S6bn9YwvwiU98gttvvx0hBEePHuVv/uZvOHny5PgHzhz/RwhxF3ChlPLfDM
```

Comment:
Again, based on the trace plots it looks like 2000 iterations is massive overkill